src/test/java/org/hyperledger/besu/ethereum/trie/verkle/TrieKeyBatchAdapterTest.java

```diff
@@ -0,0 +1,187 @@
+import org.hyperledger.besu.ethereum.trie.verkle.hasher.PedersenHasher;
+import org.hyperledger.besu.ethereum.trie.verkle.util.Parameters;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.units.bigints.UInt256;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class TrieKeyBatchAdapterTest {
+  Bytes address = Bytes.fromHexString("0x00112233445566778899aabbccddeeff00112233");
+  TrieKeyBatchAdapter adapter = new TrieKeyBatchAdapter(new PedersenHasher());
+
+  @Test
+  public void testAccountKeys() {
+    final List<Bytes32> expectedKeys = new ArrayList<>();
+    expectedKeys.add(Parameters.VERSION_LEAF_KEY);
+    expectedKeys.add(Parameters.BALANCE_LEAF_KEY);
+    expectedKeys.add(Parameters.NONCE_LEAF_KEY);
+    expectedKeys.add(Parameters.CODE_KECCAK_LEAF_KEY);
+    expectedKeys.add(Parameters.CODE_SIZE_LEAF_KEY);
+    final List<Bytes32> expectedValues =
+        List.of(
+            Bytes32.fromHexString(
+                "0x46b95e4e504b92d984c91d6f17eba4b60b904fb370818f0b6e74bc3ae5034400"),
+            Bytes32.fromHexString(
+                "0x46b95e4e504b92d984c91d6f17eba4b60b904fb370818f0b6e74bc3ae5034401"),
+            Bytes32.fromHexString(
+                "0x46b95e4e504b92d984c91d6f17eba4b60b904fb370818f0b6e74bc3ae5034402"),
+            Bytes32.fromHexString(
+                "0x46b95e4e504b92d984c91d6f17eba4b60b904fb370818f0b6e74bc3ae5034403"),
+            Bytes32.fromHexString(
+                "0x46b95e4e504b92d984c91d6f17eba4b60b904fb370818f0b6e74bc3ae5034404"));
+    final Map<Bytes32, Bytes32> map =
+        IntStream.range(0, expectedKeys.size())
+            .boxed()
+            .collect(Collectors.toMap(expectedKeys::get, expectedValues::get));
+    final Map<Bytes32, Bytes32> generatedHashes =
+        adapter.manyAccountTrieKeys(address, expectedKeys, new ArrayList<>(), new ArrayList<>());
+    assertThat(generatedHashes).containsExactlyEntriesOf(map);
+  }
+
+  @Test
+  public void testAccountKeysWithStorage() {
+    final List<Bytes32> expectedKeys = new ArrayList<>();
+    expectedKeys.add(Parameters.VERSION_LEAF_KEY);
+    expectedKeys.add(Parameters.BALANCE_LEAF_KEY);
+    expectedKeys.add(Parameters.NONCE_LEAF_KEY);
+    expectedKeys.add(Parameters.CODE_KECCAK_LEAF_KEY);
+    expectedKeys.add(Parameters.CODE_SIZE_LEAF_KEY);
+
+    final UInt256 storage = UInt256.valueOf(64);
+    expectedKeys.add(storage);
+
+    final List<Bytes32> expectedValues =
```

Comment:
think these are the expected keys, just a rename for a better comprehension
```suggestion
    final List<Bytes32> expectedKeys =
```

Reply:
done